File: cielo24_PHP/tests/JobTest.php
<?php

require_once("ActionsTest.php");
require_once("../src/cielo24/options/CaptionOptions.php");
require_once("../src/cielo24/Enums.php");

class JobTest extends ActionsTest {

    private $jobId = null;
    private $taskId = null;

    public function setUp()
    {
        parent::setUp();
        // Always start with a fresh job
        $this->jobId = $this->actions->createJob($this->apiToken, "PHP_test_job")["JobId"];
    }

    public function testOptions()
    {
        // TODO
        //$options = new CaptionOptions();
        //$options->caption_by_sentence = true;
        //$options->force_case = TextCase::UPPER;
        //$this->assertEquals("build_url=true&caption_by_sentence=true&dfxp_header=header&force_case=upper", options . ToQuery() . ToLower());
    }


    public function testCreateJob()
    {
        $result = $this->actions->createJob($this->apiToken, "test_name", "en");
        $this->assertEquals(32, strlen($result["JobId"]));
        $this->assertEquals(32, strlen($result["TaskId"]));
    }


    public function testAuthorizeJob()
    {
        $this->actions->authorizeJob($this->apiToken, $this->jobId);
    }


    public function testDeleteJob()
    {
        $this->taskId = $this->actions->deleteJob($this->apiToken, $this->jobId);
        $this->assertEquals(32, strlen($this->taskId));
    }

    public function testGetJobInfo()
    {
        $info = $this->actions->getJobInfo($this->apiToken, $this->jobId);
    }

    public function testGetJobList()
    {
        $list = $this->actions->getJobList($this->apiToken);
    }

    public function testGetElementList()
    {
        $list = $this->actions->getElementList($this->apiToken, $this->jobId);
    }

    public function testGetListOfElementLists()
    {
        $list = $this->actions->getListOfElementLists($this->apiToken, $this->jobId);
    }

    public function testGetMedia()
    {
        // Add media to job first
        $this->actions->addMediaToJobUrl($this->apiToken, $this->jobId, $this->config->sampleVideoUri);
        // Test get media
        $uri = $this->actions->getMedia($this->apiToken, $this->jobId);
        if (!filter_var($uri, FILTER_VALIDATE_URL)) {
            $this->fail("Did not receive a URL back.");
        }
    }

    public function testGetTranscript()
    {
        $this->actions->getTranscript($this->apiToken, $this->jobId);
    }

    public function testGetCaption()
    {
        $this->actions->getCaption($this->apiToken, $this->jobId, CaptionFormat::SRT);
    }

    public function testGetCaptionBuildUrl()
    {
        $options = new CaptionOptions();
        $options->build_url = true;
        $response = $this->actions->getCaption($this->apiToken, $this->jobId, CaptionFormat::SRT, $options);
        if (!filter_var($response, FILTER_VALIDATE_URL)) {
            $this->fail("Did not receive a URL back.");
        }
    }

    public function testPerformTranscription()
    {
        $this->actions->addMediaToJobUrl($this->apiToken, $this->jobId, $this->config->sampleVideoUri);
        $this->taskId = $this->actions->performTranscription($this->apiToken, $this->jobId, Fidelity::PREMIUM, Priority::STANDARD);
        $this->assertEquals(32, strlen($this->taskId));
    }

    public function testAddMediaToJobUrl()
    {
        $this->taskId = $this->actions->addMediaToJobUrl($this->apiToken, $this->jobId, $this->config->sampleVideoUri);
        $this->assertEquals(32, strlen($this->taskId));
    }

    public function testAddMediaToJobEmbedded()
    {
        $this->taskId = $this->actions->addMediaToJobEmbedded($this->apiToken, $this->jobId, $this->config->sampleVideoUri);
        $this->assertEquals(32, strlen($this->taskId));
    }

    public function testAddMediaToJobFile()
    {
        $this->taskId = $this->actions->addMediaToJobFile($this->apiToken, $this->jobId, $this->config->sampleVideoFilePath);
        $this->assertEquals(32, strlen($this->taskId));
    }
}

File: cielo24_.NET/UnitTest/JobTest.cs
using Cielo24;
using Cielo24.JSON;
using Cielo24.JSON.ElementList;
using Cielo24.JSON.Job;
using Cielo24.Options;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UnitTest
{
    [TestClass]
    public class JobTest : ActionsTest
    {
        protected Guid jobId = Guid.Empty;
        protected Guid taskId = Guid.Empty;
        protected Uri sampleVideoUri = new Uri("http://techslides.com/demos/sample-videos/small.mp4");
        protected string sampleVideoFilePath = "C:\\path\\to\\file.mp4";

        [TestInitialize]
        public void InitializeJob()
        {
            this.InitializeActions();
            // Always start with a fresh job
            this.jobId = this.actions.CreateJob(this.apiToken).JobId;
        }

        [TestMethod]
        public void testOptions()
        {
            CaptionOptions options = new CaptionOptions();
            options.CaptionBySentence = true;
            options.ForceCase = Case.upper;
            String[] array = new String[] { "build_url=true", "dfxp_header=header" };
            options.PopulateFromArray(array);
            Assert.AreEqual("build_url=true&caption_by_sentence=true&dfxp_header=header&force_case=upper", options.ToQuery().ToLower());
        }

        [TestMethod]
        public void testCreateJob()
        {
            CreateJobResult result = this.actions.CreateJob(this.apiToken, "test_name", "en");
            Assert.AreEqual(32, result.JobId.ToString("N").Length);
            Assert.AreEqual(32, result.TaskId.ToString("N").Length);
        }

        [TestMethod]
        public void testAuthorizeJob()
        {
            this.actions.AuthorizeJob(this.apiToken, this.jobId);
        }

        [TestMethod]
        public void testDeleteJob()
        {
            this.taskId = this.actions.DeleteJob(this.apiToken, this.jobId);
            Assert.AreEqual(32, this.taskId.ToString("N").Length);
        }

        [TestMethod]
        public void testGetJobInfo()
        {
            Job info = this.actions.GetJobInfo(this.apiToken, this.jobId);
        }

        [TestMethod]
        public void testGetJobList()
        {
            JobList list = this.actions.GetJobList(this.apiToken);
        }

        [TestMethod]
        public void testGetElementList()
        {
            ElementList list = this.actions.GetElementList(this.apiToken, this.jobId);
        }

        [TestMethod]
        public void testGetListOfElementLists()
        {
            List<ElementListVersion> list = this.actions.GetListOfElementLists(this.apiToken, this.jobId);
        }

        [TestMethod]
        public void testGetMedia()
        {
            // Add media to job first
            this.actions.AddMediaToJob(this.apiToken, this.jobId, this.sampleVideoUri);
            // Test get media
            Uri uri = this.actions.GetMedia(this.apiToken, this.jobId);
        }

        [TestMethod]
        public void testGetTranscript()
        {
            this.actions.GetTranscript(this.apiToken, this.jobId);
        }

        [TestMethod]
        public void testGetCaption()
        {
            this.actions.GetCaption(this.apiToken, this.jobId, CaptionFormat.SRT);
        }

        [TestMethod]
        public void testGetCaptionBuildUrl()
        {
            CaptionOptions options = new CaptionOptions(buildUri:true);
            string response = this.actions.GetCaption(this.apiToken, this.jobId, CaptionFormat.SRT, options);
            Uri uri = new Uri(response);
        }

        [TestMethod]
        public void testPerformTranscription()
        {
            this.actions.AddMediaToJob(this.apiToken, this.jobId, this.sampleVideoUri);
            this.taskId = this.actions.PerformTranscription(this.apiToken, this.jobId, Fidelity.PREMIUM, Priority.STANDARD);
            Assert.AreEqual(32, this.taskId.ToString("N").Length);
        }

        [TestMethod]
        public void testAddMediaToJobUrl()
        {
            this.taskId = this.actions.AddMediaToJob(this.apiToken, this.jobId, this.sampleVideoUri);
            Assert.AreEqual(32, this.taskId.ToString("N").Length);
        }

        [TestMethod]
        public void testAddMediaToJobEmbedded()
        {
            this.taskId = this.actions.AddEmbeddedMediaToJob(this.apiToken, this.jobId, this.sampleVideoUri);
            Assert.AreEqual(32, this.taskId.ToString("N").Length);
        }

        [TestMethod]
        public void testAddMediaToJobFile()
        {
            FileStream fs = new FileStream(this.sampleVideoFilePath, FileMode.Open);
            this.taskId = this.actions.AddMediaToJob(this.apiToken, this.jobId, fs);
            Assert.AreEqual(32, this.taskId.ToString("N").Length);
        }
    }
}
